vmdk_plugin/utils/fs/fs_linux.go

```diff
@@ -0,0 +1,352 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This is the filesystem interface for mounting volumes on a linux guest.
+
+package fs
+
```

Comment:
If refactoring its ok to add that piece now so the windows implementation can be added in seamlessly in the vmdk driver. Its only an interface definition that can use the names already in fs_linux.go. Extract those fs functions used in vmdk_driver.go and define that in an interface, instantiate the FS interface in vmdk_driver init() and thats done. Its not a major change.